schemas/public/ekscluster-kfd-v1alpha2.json

```diff
@@ -1108,29 +1108,83 @@
       },
       "required": [
         "baseDomain",
-        "dns",
         "nginx"
       ],
-      "if": {
-        "properties": {
-          "nginx": {
+      "allOf": [
+        {
+          "if": {
             "properties": {
-              "tls": {
+              "nginx": {
                 "properties": {
-                  "provider": {
-                    "const": "certManager"
+                  "type": {
+                    "const": "dual"
                   }
                 }
               }
             }
+          },
+          "then": {
+            "required": [
+              "dns"
+            ],
+            "properties": {
+              "dns": {
+                "required": [
+                  "public",
+                  "private"
+                ]
+              }
+            }
+          }
+        },
+        {
+          "if": {
+            "properties": {
+              "nginx": {
+                "properties": {
+                  "type": {
+                    "const": "single"
+                  }
+                }
+              }
+            }
+          },
+          "then": {
+            "required": [
+              "dns"
+            ],
+            "properties": {
+              "dns": {
+                "required": [
+                  "public"
```

Comment:
I don't know if it makes sense to allow users to choose between public and private in the case of a single ingress class. Maybe currently, we don't have any cases in which there is just an ingress class, and it should be private.

Reply:
Yes, currently to have the private one, the dual ingress must be enabled.